Library/Homebrew/extend/os/linux/dependency_collector.rb

```diff
@@ -0,0 +1,5 @@
+class DependencyCollector
+  def xz_dep_if_needed(tags)
+    Dependency.new("xz", tags) unless which("xz")
```

Comment:
I'm 👍 on changing the default implementation in `Library/Homebrew/dependency_collector.rb` to do this for consistency. Could also do `cvs_dep_if_needed` if desired. The only potential gotcha is `self.tar_needs_xz_dependency?`; is it a safe assumption that `xz` in the `PATH` means `tar` is built with `xz` support? CC @sjackman

Reply:
from looking at the `gnu-tar`'s `./configure` script, it looks like tar is always built with "tar" support and the only thing one can override is *which* "xz" to use. By default, tar uses the first 'xz' that appears in the PATH first.
And I don't see anything related to `xz` hard-coded in `tar`: 
```
$ strings /usr/local/Cellar/gnu-tar/1.30/bin/gtar | grep xz

$ strings /usr/local/Cellar/gnu-tar/1.30/libexec/gnubin/tar | grep xz

```